From baabf41700f7790f95007cb3c96929f4e041c6ca Mon Sep 17 00:00:00 2001
From: Devin Lin <espidev@gmail.com>
Date: Wed, 8 Jul 2026 23:21:20 -0400
Subject: [PATCH] Properly support using qqc2-desktop-style

Properly support solely using qqc2-desktop-style:
- I forgot to remove the manual setting of qqc2-breeze-style in the
settings in https://invent.kde.org/plasma/plasma-bigscreen/-/merge_requests/199, it's now removed
- Manually add a background to the sidebar component modal background,
  because qqc2-desktop-style doesn't have one
- Turn off the automatic scrollbar introduced by qqc2-desktop-style for
  ScrollView in
  the search view
---
 .../bigscreenplugin/qml/controls/SidebarOverlay.qml      | 9 +++++++++
 .../package/contents/ui/search/SearchWindow.qml          | 3 +++
 settingsapp/main.cpp                                     | 1 -
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/components/bigscreenplugin/qml/controls/SidebarOverlay.qml b/components/bigscreenplugin/qml/controls/SidebarOverlay.qml
index 7c0c8dda..a60677f2 100644
--- a/components/bigscreenplugin/qml/controls/SidebarOverlay.qml
+++ b/components/bigscreenplugin/qml/controls/SidebarOverlay.qml
@@ -62,6 +62,15 @@ QQC2.Popup {
         }
     }
 
+    QQC2.Overlay.modal: Rectangle {
+        color: Qt.rgba(0,0,0,0.4)
+        Behavior on opacity {
+            OpacityAnimator {
+                duration: Kirigami.Units.longDuration
+            }
+        }
+    }
+
     background: Item {
         Rectangle {
             id: frame
diff --git a/containments/homescreen/package/contents/ui/search/SearchWindow.qml b/containments/homescreen/package/contents/ui/search/SearchWindow.qml
index e9668009..77884663 100644
--- a/containments/homescreen/package/contents/ui/search/SearchWindow.qml
+++ b/containments/homescreen/package/contents/ui/search/SearchWindow.qml
@@ -168,6 +168,9 @@ Window {
                 Layout.maximumWidth: column.columnContentWidth
                 Layout.alignment: Qt.AlignHCenter
 
+                QQC2.ScrollBar.vertical: null
+                QQC2.ScrollBar.horizontal: null
+
                 SearchListView {
                     id: listView
                     anchors.fill: parent
diff --git a/settingsapp/main.cpp b/settingsapp/main.cpp
index 24dcdbc9..2e6ed227 100644
--- a/settingsapp/main.cpp
+++ b/settingsapp/main.cpp
@@ -22,7 +22,6 @@
 int main(int argc, char *argv[])
 {
     qputenv("PLASMA_PLATFORM", QByteArray("mediacenter"));
-    qputenv("QT_QUICK_CONTROLS_STYLE", QByteArray("org.kde.breeze"));
 
     QGuiApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
 
-- 
GitLab

